src/charm.py

```diff
@@ -34,18 +35,23 @@ def gen_pass() -> str:
 class MinioCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
+        self.minio_interface = MinioProvide(self, "minio")
         self.image = OCIImageResource(self, "oci-image")
         self.framework.observe(self.on.install, self.set_pod_spec)
         self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
         self.framework.observe(self.on.minio_relation_joined, self.send_info)
 
     def send_info(self, event):
-        secret_key = get_or_set("password", default=gen_pass)
-        event.relation.data[self.unit]["service"] = self.model.app.name
-        event.relation.data[self.unit]["port"] = self.model.config["port"]
-        event.relation.data[self.unit]["access-key"] = self.model.config["access-key"]
-        event.relation.data[self.unit]["secret-key"] = secret_key
+        secret_key = get_or_set("password", configured=None, default=gen_pass)
```

Comment:
Can this actually get updated to match the other usage of `get_or_set`?

```
secret_key = get_or_set(
    "password",
    configured=self.model.config["secret-key"],
    default=gen_pass,
)
```
I just forgot to update this one when adding that argument